### PR TITLE
SP-60466: introduce BitmapMissingFonts flag

### PR DESCRIPTION
**Pull request by emirozer**

### CODE REVIEW

File: OfficeToPDF/PowerpointConverter.cs [hunk 1]
```diff
@@ -156,12 +156,14 @@ public static new int Convert(String inputFile, String outputFile, Hashtable opt
...
                     }
                     
+                   
                     // Sometimes, presentations can have restrictions on them that block
                     // access to the object model (e.g. fonts containing restrictions).
                     // If we attempt to access the object model and fail, then try a more
                     // sneaky method of getting the presentation - create an empty presentation
                     // and insert the slides from the original file.
                     Fonts fonts = null;
+                    Boolean refFonts = (Boolean)options["powerpoint_ref_fonts"];
```
Comment (becoded) [inline OfficeToPDF/PowerpointConverter.cs]: PP: follow the same style like it was used in word (https://github.com/showpad/OfficeToPDF/blob/master/OfficeToPDF/WordConverter.cs#L63)
`Boolean bitmapMissingFonts = !(Boolean)options["powerpoint_ref_fonts"];`

File: OfficeToPDF/PowerpointConverter.cs [hunk 2]
```diff
@@ -195,7 +197,7 @@ public static new int Convert(String inputFile, String outputFile, Hashtable opt
                     {
                         try
                         {
-                            activePresentation.ExportAsFixedFormat(outputFile, PpFixedFormatType.ppFixedFormatTypePDF, quality, MSCore.MsoTriState.msoFalse, PpPrintHandoutOrder.ppPrintHandoutVerticalFirst, printType, MSCore.MsoTriState.msoFalse, null, PpPrintRangeType.ppPrintAll, "", includeProps, true, includeTags, true, pdfa, Type.Missing);
+                            activePresentation.ExportAsFixedFormat(outputFile, PpFixedFormatType.ppFixedFormatTypePDF, quality, MSCore.MsoTriState.msoFalse, PpPrintHandoutOrder.ppPrintHandoutVerticalFirst, printType, MSCore.MsoTriState.msoFalse, null, PpPrintRangeType.ppPrintAll, "", includeProps, true, includeTags, refFonts, pdfa, Type.Missing);
```
Comment (becoded) [inline OfficeToPDF/PowerpointConverter.cs]:
```suggestion
                            activePresentation.ExportAsFixedFormat(outputFile, PpFixedFormatType.ppFixedFormatTypePDF, quality, MSCore.MsoTriState.msoFalse, PpPrintHandoutOrder.ppPrintHandoutVerticalFirst, printType, MSCore.MsoTriState.msoFalse, null, PpPrintRangeType.ppPrintAll, "", includeProps, true, includeTags, bitmapMissingFonts, pdfa, Type.Missing);

```

File: OfficeToPDF/Program.cs [hunk 3]
```diff
@@ -1087,6 +1088,10 @@ Office application
   /powerpoint_output <type> - Controls what is generated by output. Possible values are slides, notes,
                               outline, build_slides, handouts and multi-page handouts using handout2,
                               handout3, handout4, handout6 and handout9. The default is slides.   
+  /powerpoint_ref_fonts     - Specifies whether to include a bitmap of the text. If False, the font is referenced,
```
Comment (becoded) [inline OfficeToPDF/Program.cs]: You could use the same description as used for `/word_ref_fonts` as it doing the same thing.

File: OfficeToPDF/PowerpointConverter.cs [hunk 4]
```diff
@@ -156,12 +156,14 @@ public static new int Convert(String inputFile, String outputFile, Hashtable opt
...
                     }
                     
+                   
                     // Sometimes, presentations can have restrictions on them that block
                     // access to the object model (e.g. fonts containing restrictions).
                     // If we attempt to access the object model and fail, then try a more
                     // sneaky method of getting the presentation - create an empty presentation
                     // and insert the slides from the original file.
                     Fonts fonts = null;
+                    Boolean bitmapMissingFonts = (Boolean)options["powerpoint_ref_fonts"];
```
Comment (becoded) [inline OfficeToPDF/PowerpointConverter.cs]:
```suggestion
                    Boolean bitmapMissingFonts = !(Boolean)options["powerpoint_ref_fonts"];

```

Comment (becoded) [reply OfficeToPDF/PowerpointConverter.cs]: Keep the default behaviour by using here `!`

File: docs/Documentation.html [hunk 5]
```diff
@@ -161,6 +161,11 @@ <h3>Command line switches</h3>
 <td>when fonts are not available, a reference to the font is used in the generated PDF rather than a bitmapped version. The default is for a bitmap of the text to be used</td>
 </tr>
 <tr>
+<td>/powerpoint_ref_fonts</td>
+<td>Specifies whether to include a bitmap of the text. Set this parameter to True when font licenses don't permit a font to be embedded in the PDF file. If False, the font is referenced, and the viewer's computer substitutes an appropriate font if the authored one is not available. Default value is False.</td>
```
Comment (becoded) [inline docs/Documentation.html]: PP: match readme with code

File: OfficeToPDF/ExcelConverter.cs [hunk 6]
```diff
@@ -332,20 +332,6 @@ public static new int Convert(String inputFile, String outputFile, Hashtable opt
                                 sheetForConversionIdx = wsIdx;
                             }
 
-                            if (showHeadings && ws is _Worksheet)
```
Comment (becoded) [inline OfficeToPDF/ExcelConverter.cs:335]: Q: are we sure this needs to be removed?
Seems related with our previous changes:
https://github.com/cognidox/OfficeToPDF/pull/63/files#diff-4162be89b015b8330e5a2e775dcccc799a44f03797eaf946042200e477d0e4f1R335

Comment (emirozer) [reply OfficeToPDF/ExcelConverter.cs:335]: There was a small mistake in the merge conflict fix this this if block was duplicated. The actual is there the duplicate is removed. (below from this branch):

image

**Comment by becoded** [comment]
👍

**Pull request merged**